### PR TITLE
Windows support (Test on more platforms)

### PR DESCRIPTION
**Pull request by LukeShu**
Three things are happening here:
 - Have CI run the tests on `GOOS=windows` and `GOOS=darwin` (not just `GOOS=linux`)
 - Update the tests to work on `GOOS=windows` and `GOOS=darwin`.
 - There only turned out to be 1 portability problem in the actual code: `dexec` soft shutdown didn't work on Windows; so fix that (Windows doesn't have SIGINT; emulate it with syscall.CTRL_BREAK_EVENT).

IDK what's going on with coveralls flopping between whether it wants to turn green or not.  This PR *does* lower the reported test coverage percentage (from  87.932% to 86.907%) so it _should_ be turning red.  But the reason the percentage is going down is because of the added `if runtime.GOOS == "windows" { … }` blocks; the coverage report is based on the `GOOS=linux` run, so adding any code that doesn't run on `GOOS=linux` will make the reported percentage go down, even if that code is covered.

### CODE REVIEW

File: dexec/cmd.go [hunk 1]
```diff
@@ -158,6 +165,19 @@ func (c *Cmd) Start() error {
...
+	// On Windows when you send "os.Interrupt" (really syscall.CTRL_BREAK_EVENT) it goes to
+	// *all* processes in the process group.  So if we're going to be sending it ourselves, put
+	// the child process in its own group.  However, we'll also need to propegate it if
+	// something else sends it to our whole group, for consistency with the vanilla behavior.
+	var windowsInterruptCh chan os.Signal
+	if runtime.GOOS == "windows" && c.ctx != dcontext.HardContext(c.ctx) {
+		windowsInterruptCh := make(chan os.Signal, 10)
+		signal.Notify(windowsInterruptCh, os.Interrupt)
+		procAttr := sysProcAttrForNewGroup
+		c.SysProcAttr = &procAttr
```
Comment (thallgren) [inline dexec/cmd.go:178]: I quess a check is needed here to verify that `SysProcAttr` isn't already set by the caller and then just add the flag if it is?

**Review by thallgren — CHANGES_REQUESTED** [review]

File: dtime/sleepwithcontext_test.go [hunk 2]
```diff
@@ -57,8 +58,15 @@ func TestSleep(t *testing.T) {
 			start := time.Now()
 			SleepWithContext(ctx, tcinfo.Arg)
 			actual := time.Since(start)
-			assertDurationEq(t, tcinfo.Expected, actual,
-				time.Second/100)
+
+			slop := 10 * time.Millisecond
```
Comment (donnyyung) [inline dtime/sleepwithcontext_test.go:62]: Dumb question, does `slop` stand for something or is it just the word that was used to reference for a time buffer of sorts?

Comment (LukeShu) [reply dtime/sleepwithcontext_test.go:62]: It's just the word.

File: dexec/cmd.go (hunk 1 above)
Comment (LukeShu) [reply dexec/cmd.go:178]: Ugg, I definitely under-thought about this.

**Pull request closed without merging**